Checkout fails on pull request closed event

Hello everyone,

So, I have a `.github/workflow/docker.yml` like this:

```
name: Maybe update Docker image

on:
  pull_request:
    types: [closed]

jobs:
  maybe_update:
    runs-on: ubuntu-latest

    if: github.event.pull_request.merged == true

    steps:
    - name: Checkout
      uses: actions/checkout@v2
```

Which is pretty basic. The goal is to run this on master after a pull request is merged. It does find the lastest commit on master (be3c9f9), as seen here:

```
/usr/bin/git config --local gc.auto 0
/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
/usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin be3c9f9d09e54097321f92f452923a91435b63ac
```
But when it comes to checking out, it cant:

```
From https://github.com/...
 * branch            be3c9f9d09e54097321f92f452923a91435b63ac -> FETCH_HEAD
/usr/bin/git branch --list --remote origin/master
/usr/bin/git tag --list master
Removed matchers: 'checkout-git'
##[error]A branch or tag with the name 'master' could not be found
##[error]Node run failed with exit code 1
```

We also tried tying it explicitly to master, like this:

```
name: Maybe update Docker image

on:
  pull_request:
    branches: [master]
    types: [closed]    
```

But the same problem occurs. Any ideas? (:

cheers

--- a/__test__/input-helper.test.ts
+++ b/__test__/input-helper.test.ts
@@ -75,6 +75,20 @@ describe('input-helper tests', () => {
     expect(settings.repositoryPath).toBe(gitHubWorkspace)
   })
 
+  it('qualifies ref', () => {
+    let originalContext = mockGitHub.context
+    try {
+      mockGitHub.context = {...originalContext} // Shallow clone
+      mockGitHub.context.ref = 'some-unqualified-ref'
+      const settings: ISourceSettings = inputHelper.getInputs()
+      expect(settings).toBeTruthy()
+      expect(settings.commit).toBe('1234567890123456789012345678901234567890')
+      expect(settings.ref).toBe('refs/heads/some-unqualified-ref')
+    } finally {
+      mockGitHub.context = originalContext
+    }
+  })
+
   it('requires qualified repo', () => {
     inputs.repository = 'some-unqualified-repo'
     assert.throws(() => {

--- a/src/input-helper.ts
+++ b/src/input-helper.ts
@@ -61,6 +61,12 @@ export function getInputs(): ISourceSettings {
     if (isWorkflowRepository) {
       result.ref = github.context.ref
       result.commit = github.context.sha
+
+      // Some events have an unqualifed ref. For example when a PR is merged (pull_request closed event),
+      // the ref is unqualifed like "master" instead of "refs/heads/master".
+      if (result.commit && result.ref && !result.ref.startsWith('refs/')) {
+        result.ref = `refs/heads/${result.ref}`
+      }
     }
 
     if (!result.ref && !result.commit) {

